shielded-pool/src/component.rs

```diff
@@ -481,18 +513,38 @@ pub trait View: StateExt {
             .await
     }
 
-    async fn set_nct_anchor(&self, height: u64, anchor: merkle::Root) {
-        tracing::debug!(?height, ?anchor, "writing anchor");
+    async fn set_nct_anchor(
+        &self,
+        height: u64,
+        nct_anchor: merkle::Root,
+        tct_anchor: penumbra_tct::Root,
+    ) {
+        tracing::debug!(?height, ?nct_anchor, "writing anchor");
 
         // Write the NCT anchor both as a value, so we can look it up,
         self.put_domain(
             format!("shielded_pool/nct_anchor/{}", height).into(),
-            anchor.clone(),
+            nct_anchor.clone(),
+        )
+        .await;
+        // and as a key, so we can query for it.
+        self.put_proto(
+            format!("shielded_pool/valid_anchors/{}", nct_anchor).into(),
+            // We don't use the value for validity checks, but writing the height
+            // here lets us find out what height the anchor was for.
+            height,
+        )
+        .await;
+
+        // Write the TCT anchor both as a value, so we can look it up,
+        self.put_domain(
+            format!("shielded_pool/tct_anchor/{}", height).into(),
+            tct_anchor,
         )
         .await;
         // and as a key, so we can query for it.
         self.put_proto(
-            format!("shielded_pool/valid_anchors/{}", anchor).into(),
+            format!("shielded_pool/valid_tct_anchors/{}", tct_anchor).into(),
```

Comment:
I think we want to roll back all the changes to this method, and leave it as-is, then, when we execute the switch-over, we'll swap
```
nct_anchor: merkle::Root
```
for
```
nct_anchor: penumbra_tct::Root
```
along with the other changes we'll need to make to the client protocol.

Reply:
Makes sense!